ext/spl/spl_directory.c

```diff
@@ -651,14 +651,17 @@ static inline HashTable *spl_filesystem_object_get_debug_info(zend_object *objec
 		pnstr = spl_gen_private_prop_name(spl_ce_SplFileInfo, "fileName", sizeof("fileName")-1);
 		path = spl_filesystem_object_get_path(intern);
 
-		if (ZSTR_LEN(path) && ZSTR_LEN(path) < ZSTR_LEN(intern->file_name)) {
+		if (path && ZSTR_LEN(path) && ZSTR_LEN(path) < ZSTR_LEN(intern->file_name)) {
+			/* +1 to skip the trailing / of the path in the file name */
```

Comment:
Wouldn't that be a leading slash?

Reply:
Not sure if this behaves differently on a windows machine, but `\SplFileInfo` IMHO always only worked properly with forward slashes. See https://3v4l.org/lsPaS.

Reply:
> Wouldn't that be a leading slash?

Huuum maybe it's wording, I used trailing as in ``'foo/bar.baz'`` the path is ``foo`` so we need to skip the slash just after ``foo`` therefore trailing the path?

Reply:
> Not sure if this behaves differently on a windows machine, but `\SplFileInfo` IMHO always only worked properly with forward slashes. See https://3v4l.org/lsPaS.

Great, more reasons for me to hate SPL :|